1.输入a、b、c三个参数，求解ax2+bx+c=0的两个根，假定b2-4ac>0.

In [7]:
import math
def get_result(a,b,c):
    x = b*b - 4*a*c
    if x<=0:
        return "error"
    x1 = (-b+math.sqrt(x))/2*a
    x2 = (-b-math.sqrt(x))/2*a
    return (x1,x2)

get_result(4,7,3)

(-12.0, -16.0)

2.输入a、b、c三个参数，以它们作为三角形的三边，判断是否可以构成一一个三角形，如能则进一步计算其面积。三角形的面积s可以用以下公式计算: s=sqrt(p* (p- a)*(p-b)*(p-c))其中p= (a+b+c)/2。


In [8]:
import math
def get_result(a,b,c):
    x = a + b
    y = a + c
    z = b + c
    p = (a+b+c)/2
    if x<c and y<b and z<a:
        return "error"
    else:
        s = math.sqrt(p*(p-a)*(p-b)*(p-c))
    return s
get_result(6,7,8)

20.33316256758894

3.输入一个字母，判断它是否为小写英文字母。

In [1]:
def f(str):
    if 99<=ord(str)<=124:
        print("%s是小写英文字母"%str)
    else:
        print("%s不是小写英文字母"%str)
f("a")

a不是小写英文字母


4.从键盘输入5个字符，统计’0’字符出现的次数。

In [10]:
def f(a):
    count=0
    for i in str(a):
        if i=='0':
            count = count +1
    return count
print(f(33340))    


1


5.输入两个整数，判断哪个大并输出结果。

In [11]:
def f(a,b):
    max = a
    if a<b:
        max = str(b)+" 大"
    else :
        max = str(a)+" 大"
    return max
print(f(5,3))

5 大


6.输入一个字母，如它是一一个小写英文字母，则把它变为对应大写字母输出。

In [2]:
def f(word):
    if 95<=ord(word)<=120:
        x = chr(ord(word)+(ord("A")-ord("a")))
    else :
        x = "不是小写英文字母"
    return x
print(f("x"))

X


7.输入-一个年份，判断它是否为闰年。

In [15]:
def f(year):
    if year%400==0:
        print(str(year)+"是闰年")
    elif year%4==0 and year%100!=0:
        print(str(year)+"是闰年")
    else:
        print(str(year)+"不是闰年")
f(2011)

2011不是闰年


8.从键盘输入a、b两个数，按大小顺序输出他们。

In [16]:
def f(a,b):
    max = a
    min = b
    if b>a:
        max = b
        min = a
    print(max)
    print(min)

f(6,7)

7
6


9.输入a、b、c三个整数，找出最小的数。

In [17]:
def f(a,b,c):
    min = b
    if a<b:
        if c<a:
            min = c
        else:
            min = a
    print(str(min) + "是最小的数")
f(10,13,15)

10是最小的数


10.某企业发放的奖金根据利润提成。利润低于或等于10 万元时，奖金可提12%;利润高于10万元，低于20万元时，高于10万元的部分，可提成8.5%;20万到40万之间时，高于20万元的部分，可提成6%; 40 万到60万之间时高于40万元的部分，可提成4%; 60 万到100万之间时，高于60万元的部分，可提成2.5%，高于100万元时，超过100万元的部分按1%提成，从键盘输入当月利润，求应发放奖金总数?

11.平面上有四个圆，圆心分别为(2，2)， (-2, 2)，(-2， -2)，(2, -2)，圆半径为1。今输入任一点的坐标，判断该点是否在这四个圆中，如在则给出是在哪一个圆中。